client/test/message_action_creator.spec.js

```diff
@@ -0,0 +1,246 @@
+  });
+
+  after(() => {
+    mockeryUtils.clean();
+  });
+
+
+  beforeEach(() => {
+    if (spyDispatcher === undefined) {
+      spyDispatcher = sinon.spy(Dispatcher, 'dispatch');
+    }
+  });
+
+  afterEach(() => {
+    spyDispatcher.reset();
+  });
+
+
+  describe('Methods', () => {
+    it('receiveRawMessages', () => {
+      let value = ['msg1', 'msg2', 'msg3'];
+      const result = { type: ActionTypes.RECEIVE_RAW_MESSAGES, value };
+      MessageActionCreator.receiveRawMessages(value);
+
+      assert.equal(spyDispatcher.callCount, 1);
+      assert.isTrue(spyDispatcher.calledWith(result));
+     });
+
+
+     it('receiveRawMessage', () => {
+       let value = 'msg1';
+       const result = { type: ActionTypes.RECEIVE_RAW_MESSAGE, value };
+       MessageActionCreator.receiveRawMessage(value);
+
+       assert.equal(spyDispatcher.callCount, 1);
+       assert.isTrue(spyDispatcher.calledWith(result));
+     });
+
+
+     it('displayImages', () => {
+      const messageID = 'plop';
+
+      let result = {
+        type: ActionTypes.SETTINGS_UPDATE_REQUEST,
+        value: { messageID, displayImages: true }
+      };
+
+      MessageActionCreator.displayImages({ messageID, displayImages: undefined });
+      assert.equal(spyDispatcher.callCount, 1);
+      assert.isTrue(spyDispatcher.calledWith(result));
+
+      result.value.displayImages = false;
+      MessageActionCreator.displayImages({ messageID, displayImages: false });
+      assert.equal(spyDispatcher.callCount, 2);
+      assert.isTrue(spyDispatcher.calledWith(result));
+
+      result.value.displayImages = true;
+      MessageActionCreator.displayImages({ messageID, displayImages: true });
+      assert.equal(spyDispatcher.callCount, 3);
+      assert.isTrue(spyDispatcher.calledWith(result));
+    });
+
+
+    describe('send', () => {
+      let spySend;
+      let callback;
+      let action = 'mon action';
+      const message = { conversationID: 'plop', text: 'coucou' };
+
+
+      beforeEach(() => {
+        if (spySend === undefined) {
+          spySend = sinon.spy(XHRUtils, 'messageSend');
+        }
+      });
+
+      afterEach(() => {
+        spySend.reset();
+      });
+
+
+      it('should MESSAGE_SEND_SUCCESS dispatched', () => {
+        const type = ActionTypes.MESSAGE_SEND_SUCCESS;
+
+        MessageActionCreator.send(action, message);
+
+        assert.equal(spySend.callCount, 1);
+
+        callback = spySend.getCall(0).args[1];
+        assert.equal(typeof callback, 'function');
+
+        // Dis is supposed to be called twice
+        // 1rst one for REQUEST
+        // Last one for SUCCESS
+        assert.equal(spyDispatcher.callCount, 2);
+
+        let args = spyDispatcher.getCall(1).args[0];
+        assert.equal(args.type, type);
+
+        let res = { conversationID: 'plop', text: 'coucou', html: 'coucou'};
+        assert.equal(args.value.action, action);
+        _.each(JSON.parse(args.value.message), (value, key) => {
```

Comment:
Seen from here, those 2 are strictly equals and `deepEqual` shoudl do the trick, nope?